### ISSUE TITLE
Action Links

### ISSUE THREAD

**Issue by gabestein** [pubpub/platform#794]
## Motivation

So users can see effects of their actions on sites off of PubPub
So authors can preview what their pub will look like after they make changes

## Requirements

- When configuring action, allows admins to specify a link that will be opened
- Allows admins to use pub fields as variables in the link (e.g. slug, to load a page of a static site)
- After a successful action run, opens the specified link in a new tab

## Open Question
Is this its own action, or is it an option to open a link post-run that's added to all actions?

**Comment by tefkah** [comment]
I don't think this should be it's own action if we want these links to open automatically, as we aren't able to sequentially link actions (and aren't planning on doing that afaik).
Additionally, actions only run on the server, while this is client behavior. It therefore makes more sense to me that this is generic "post-action" behavior that could be configured on a per action basis.

Additional questions:
- What happens when the person who initiates the action navigates away before it finishes?

Other solutions:
- We have something like an "action output" we can display in the action logs. we could make that display a link instead.
- We could send action output via email to the initiator instead (or in addition/when they navigate away)

**Comment by tefkah** [comment]
Also: do we really need the link to auto open? We could also just show an "open link" button in the toast. i feel like that could lead to some more general behavior (by default: go to action log; other option: go to pub)

**Comment by gabestein** [comment]
Hrm, reading this I'm not sure I fully explained it.

One use-case for this is simply being able to add a button to a pub that loads a URL with a pub variable in it, like `https://dev-server.com/pub-preview/id`

The other is, yes, to chain a bit, such that you run a build-step and then, when finished, return a link. So maybe it's two things?

**Comment by tefkah** [comment]
Ah right, that makes sense. I'm not sure whether the first use case warrants an action on its own, I think that might be better as dedicated behavior, eg being able to specify a field (or interpolation of a field) as an "external url" or "main url" or something in a PubType (like the title), and then we render that button in select places.

I think for now you would just write that info to a field on that pub, and then go to the pub page and click that link.
Either the sitebuilding action could take care of writing that information for now using the same logic as in #780 (specify a pubfield to write to), or we would create some specific action which would allow you to interpolate other fields of a pub into a different specific pub field. Eg an action called "Create Preview URL", in which you would specify to write to `croccroc:preview-url` something like `https://dev-server.com/pub/:value{id}`.  You could set this action to run when a pub enters the `published` stage.

I personally feel like the described behavior (adding a button that open a url) is quite custom and not that trivial to add. I'd prefer it if someone would spike on this first to figure out a nice pattern for these things.

Ideally i would like us to be able to get away with just adding this behavior to the publish action for now, in that it would update a field on the pub it's run on to have the relevant url, and maybe return the url as a link in the toast once the action is finished.

**Comment by gabestein** [comment]
Thanks for this. I added #818 to the top of refining, and may try to get it in at the bottom of the sprint. Mind taking a look and providing feedback/estimate when you can?
